### PR TITLE
gRPC: Example wallet skeleton

### PR DESCRIPTION
**Pull request by rh-concordium**
## Purpose

Create the skeleton for the example iOS wallet. 

The example wallet contains the same views as its Android equivalent and logic can be incrementally added as it becomes available in the client. Thus, it implements step 1 of this jira: https://concordium.atlassian.net/browse/CBW-1570

## Changes

Created wallet skeleton with same screens as on Android (although with an empty implementation for most screens for now). The only logic is seed phrase validation and a few tests thereof was added. 

Sidenote: We might want to consider hiding away the Grpc details from the client, like `GRPCChannelPool`. This allows clients (e.g., the example wallet) to only import ConcordiumSwiftSDK and not GRPC and NIOPosix as well, when creating client instances.

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [X] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [X] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: ExampleWallet/ExampleWallet/SeedPhraseValidator.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+struct RealSeedPhraseValidator: SeedPhraseValidator {
+    func isValid(seedPhrase: String) -> Bool {
+        do {
+            try Mnemonic.validate(mnemonic: seedPhrase)
+            return true
+        } catch {
+            return false
+        }
+    }
+}
```
Comment (bisgardo) [inline ExampleWallet/ExampleWallet/SeedPhraseValidator.swift:17]: Not I sure I see the utility of this. Why not just use `Mnemonic.validate` directly? That also seems to provide a useful error message.

File: Sources/ConcordiumSwiftSDK/Client.swift [hunk 2]
```diff
@@ -3,14 +3,14 @@ import GRPC
 import NIOCore
 import NIOPosix
 
-public class Client {
+class Client {
```
Comment (bisgardo) [inline Sources/ConcordiumSwiftSDK/Client.swift:6]: Why is this changed?

File: Sources/ConcordiumSwiftSDK/Client.swift [hunk 3]
```diff
@@ -20,21 +20,13 @@ public class Client {
...
         )
     }
-
-    public func getAccountInfo(of account: AccountIdentifier, at block: BlockIdentifier) async throws -> AccountInfo {
-        var req = Concordium_V2_AccountInfoRequest()
-        req.accountIdentifier = account.toGrpcType()
-        req.blockHash = block.toGrpcType()
-        let res = try await grpc.getAccountInfo(req).response.get()
-        return try .fromGrpcType(res)
-    }
```
Comment (bisgardo) [inline Sources/ConcordiumSwiftSDK/Client.swift:39]: And this?

File: examples/get-cryptographic-parameters/Package.swift
Comment (bisgardo) [inline examples/get-cryptographic-parameters/Package.swift:1]: Why is this deleted?

**Review by bisgardo — CHANGES_REQUESTED** [review]

File: .swift-version
[the remark this answers is not in the document]
Comment (rh-concordium) [reply .swift-version:1]: I suspect that the hail mary to get out of that weird git state resulted in this. 

Basically, nothing under the ConcordiumSwiftSDK should have been changed in this PR, so I'll just ensure that those files use the `main` version

File: ExampleWallet/ExampleWallet/SeedPhraseValidator.swift (hunk 1 above)
Comment (rh-concordium) [reply ExampleWallet/ExampleWallet/SeedPhraseValidator.swift:17]: Two reasons: 1) To mimic the Android implementation, 2) To provide an abstraction layer, so that we can change the underlying implementation to another library later (BitcoinKit or whatever), without affecting the rest of the app.

File: ExampleWallet/ExampleWallet.xcodeproj/xcuserdata/renehansen.xcuserdatad/xcschemes/xcschememanagement.plist
[the remark this answers is not in the document]
Comment (rh-concordium) [reply ExampleWallet/ExampleWallet.xcodeproj/xcuserdata/renehansen.xcuserdatad/xcschemes/xcschememanagement.plist:1]: Removed

File: Sources/ConcordiumSwiftSDK/Client.swift (hunk 2 above)
Comment (rh-concordium) [reply Sources/ConcordiumSwiftSDK/Client.swift]: Bogus git

File: examples/get-cryptographic-parameters/Package.swift
Comment (rh-concordium) [reply examples/get-cryptographic-parameters/Package.swift:1]: Bogus git - the folder is back

**Review by bisgardo — APPROVED** [review]
Looks reasonable! I tried to run it ran into some issues that I think are with my PC. For one it complained that it couldn't find `ConcordiumSwiftSdk` package.

I assume you verified that it works, in which case it's good enough to start with :)

File: ExampleWallet/ExampleWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    },
+    {
+      "identity" : "bitcoinkit",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/yenom/BitcoinKit",
+      "state" : {
+        "branch" : "master",
+        "revision" : "74bd7642dfb3f6af6baf242c7bb084af6ee3ae1b"
+      }
+    },
```
Comment (bisgardo) [inline ExampleWallet/ExampleWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:38]: This lib is not used right?

Comment (rh-concordium) [reply ExampleWallet/ExampleWallet.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:38]: You're right - I'll see about nuking it

File: ExampleWallet/ExampleWalletTests/ExampleWalletTests.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
+@testable import ExampleWallet
+import XCTest
+
+class ExampleWalletTests: XCTestCase {
+    var sut: SeedPhraseValidator!
```
[the remark this answers is not in the document]
Comment (rh-concordium) [reply ExampleWallet/ExampleWalletTests/ExampleWalletTests.swift:5]: That's an oversight on my part. It was of course supposed to be used in both tests, and not be created locally in one of them

File: ExampleWallet/ExampleWallet/SeedPhraseValidator.swift (hunk 1 above)
Comment (rh-concordium) [reply ExampleWallet/ExampleWallet/SeedPhraseValidator.swift:17]: Yeah, it might be overkill to wrap the MnemonicSwift package if it turns out that this will be the only usage in the whole example app. 
However, in general I think it is a good practice to wrap third party libraries to get a looser coupling, and make it easier to switch the actual implementation / library. 
I also suspect that this interface will probably evolve slightly to wrap a few more Cryptographic methods (which may in fact be implemented by different libraries) - in that case it should probably also get a new name.

**Comment by rh-concordium** [comment]
> Looks reasonable! I tried to run it ran into some issues that I think are with my PC. For one it complained that it couldn't find `ConcordiumSwiftSdk` package.
> 
> I assume you verified that it works, in which case it's good enough to start with :)

Hmm... that's weird. We can take a look at it tomorrow. In order to run the example wallet, I open the ExampleWallet project directly, rather than opening up the project at the root of the concordium-swift-sdk folder.
